src/DataStructures/Tensor/Expressions/TensorExpression.hpp

```diff
@@ -70,3 +296,52 @@ template <typename Derived, typename DataType, typename Symm,
 TensorExpression<Derived, DataType, Symm, tmpl::list<Indices...>,
                  ArgsList<Args...>>::~TensorExpression() = default;
 /// @}
+
+namespace TensorExpressions {
+namespace detail {
+/// @{
+/// \brief The maximum number of arithmetic tensor operations allowed in a
+/// `TensorExpression` subtree before having it be a splitting point in the
+/// overall RHS expression, according to the data type held by the `Tensor`s in
+/// the expression
+///
+/// \details
+/// To enable splitting for `TensorExpression`s with a different data type,
+/// define a new variable below like `max_num_ops_in_datavector_sub_expression`
+/// for your data type, then update the control flow in
+/// `max_num_ops_in_sub_expression_helper`.
+///
+/// Before defining a max operations cap for some data type, the change should
+/// be first justified by benchmarking before and after introducing the new cap.
+static constexpr size_t max_num_ops_in_datavector_sub_expression = 8;
+/// @}
+
+/// \brief Helper struct for getting the maximum number of arithmetic tensor
+/// operations allowed in a `TensorExpression` subtree before having it be a
+/// splitting point in the overall RHS expression, according to the `DataType`
+/// held by the `Tensor`s in the expression
+///
+/// \tparam DataType the type of the data being stored in the `Tensor`s in the
+/// `TensorExpression`
+template <typename DataType>
+struct max_num_ops_in_sub_expression_helper {
+  // Splitting is only enabled for expressions when DataType == DataVector
+  // because benchmarking has shown it to be beneficial. To enable splitting
+  // for other data types, define a new static variable like
+  // `max_num_ops_in_datavector_sub_expression` for the data type of interest,
+  // then update the control flow below
+  static constexpr size_t value = std::is_same_v<DataType, DataVector>
```

Comment:
I think we have an `is_vector_impl` type alias somewhere (I can try to find it if you want). You could use that to generalize the splitting logic to anything that is a Blaze vector we've wrapped :)

Reply:
Because TEs only work for `double` and `DataVector` right now and because I haven't benchmarked to see if, say, `ComplexDataVector` requires a different number, I think maybe I should keep it specific to `DataVector`, for now. What do you think?

Reply:
Yea, that sounds good to me :)